test/Microsoft.SqlTools.ServiceLayer.UnitTests/Profiler/ProfilerServiceTests.cs

```diff
@@ -254,21 +254,14 @@ public async Task TestStoppedSessionNotification()
             profilerService.ConnectionServiceInstance = TestObjects.GetTestConnectionService();
             ConnectionInfo connectionInfo = TestObjects.GetTestConnectionInfo();
             profilerService.ConnectionServiceInstance.OwnerToConnectionMap.Add(testUri, connectionInfo);
+            mockSession.SetupProperty(p => p.ConnDetails, connectionInfo.ConnectionDetails);
+            mockSession.SetupProperty(p => p.Session, new Session(null, "profiler_uri"));
 
             // start monitoring test session
             profilerService.SessionMonitor.StartMonitoringSession(testUri, mockSession.Object);
 
-            // wait for polling to finish, or for timeout
-            System.Timers.Timer pollingTimer = new System.Timers.Timer();
-            pollingTimer.Interval = 10000;
-            pollingTimer.Start();
-            bool timeout = false;
-            pollingTimer.Elapsed += new System.Timers.ElapsedEventHandler((s_, e_) => { timeout = true; });
-            while (sessionStopped == false && !timeout)
-            {
-                Thread.Sleep(250);
-            }
-            pollingTimer.Stop();
+            // stop session with fake stop error.
```

Comment:
I'm not sure I understand how this is a "fake stop error". There's no error involved at all. And if you're simulating an error it would be better to mock the stream and then use the mock to send an error that way instead.